python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -770,10 +770,17 @@ def process_graph(args):
                           skip_nonexist_edges=args.skip_nonexist_edges)
     sys_tracker.check('Process the edge data')
     num_nodes = {ntype: len(raw_node_id_maps[ntype]) for ntype in raw_node_id_maps}
+
+    os.makedirs(args.output_dir, exist_ok=True)
+
     if args.output_conf_file is not None:
-        # Save the new config file.
-        with open(args.output_conf_file, "w", encoding="utf8") as outfile:
-            json.dump(process_confs, outfile, indent=4)
+        outfile_path = args.output_conf_file
+    else:
+        outfile_path = os.path.join(args.output_dir, 'data_transform_new.json')
```

Comment:
Do we need to check whether 'data_transform_new.json' is already there? 
Are we going to overwrite it?

Reply:
If users do not provide this argument, we generate it by default. So the default behavior is to overwrite the existing one.

We can check the existence of the file, but the next operation is complex. Options are
- rename the existing one, and the problem is how to rename the existing one.
- rename the new file, and the problem is same, i.e., how to rename the new file.

If we follow some rules to rename files, we need to check if renamed files exist too, like MacOS does. But Python os library does not have this function.

Another potential issue might be the permission to rename existing files. We may confront some error when we cannot rename existing files.

So I would vote for overwriting the existing file.

Reply:
Add new warning message when overwriting happens.